### DISCUSSION TITLE
Importing Shodan  scan results

### DISCUSSION THREAD

**Discussion by hackykhaki** [DefectDojo/django-DefectDojo#5987] [Q&A] [answered]
Hi, I'm finding a way to import CSV scan results from Shodan in to DefectDojo. Is it because what I get from Shodan is commonly same (and more) from scan results from NMAP or Nessus hence there is no parser for it? Is this not a common use case?

**Comment by manuel-sommer** [comment, marked as answer]
Hi @hackykhaki ,
If a parser is missing for any vulnerability resource, you have two ways:

- submit a pull request. To get started take a look here: https://defectdojo.github.io/django-DefectDojo/contributing/how-to-write-a-parser/ 
- open up an issue with an anonymized sample file and details about the parser. A contributer will probably pick it up and submit a pull request for you.

Does this help you? 

Best,

Manuel
